docs/_docs/install/settings.md

```diff
@@ -6,73 +6,119 @@ toc: false
 
 # Settings
 
-See that folder called [settings](https://github.com/singularityhub/sregistry/blob/master/shub/settings)? inside are a bunch of different starting settings for the application. We will change them in these files before we start the application. There are actually only two files you need to poke into, generating a `settings/secrets.py` from our template [settings/dummy_secrets.py](https://github.com/singularityhub/sregistry/blob/master/shub/settings/dummy_secrets.py) for application secrets, and [settings/config.py](https://github.com/singularityhub/sregistry/blob/master/shub/settings/config.py) to configure your database and registry information.
-
-## Secrets
-
-There should be a file called `secrets.py` in the shub settings folder (it won't exist in the repo, you have to make it), in which you will store the application secret and other social login credentials.
-
-An template to work from is provided in the settings folder called `dummy_secrets.py`. You can copy this template:
-
-```bash
-cp shub/settings/dummy_secrets.py shub/settings/secrets.py
+As of version `2.0.0` we have one core [settings.py](https://github.com/singularityhub/sregistry/blob/master/shub/settings.py) file
+that can be populated in several ways:
+
+ - Directly in the file (e.g., if you build your own base container)
+ - From the environment (described below with a `SREGISTRY_` prefix
+ - From a config file, `settings.yaml` in the root directory of your server OR custom set at `SREGISTRY_SETTINGS_FILE`
+ 
+Order of preference or variables honored is:
+
+ 1. secrets.py
```

Comment:
The way it is coded now, the order is:
1. secrets.py
2. settings.yaml
3. the environment
4. defaults directly in settings.py

Which I think is the right order, because we:
1. set the defaults
2. override any values from the env
3. override any values from settings.yaml
4. only override values already set in the file from `secrets.yaml`.

Reply:
Environment (starting on line 226) actually comes before settings.yaml (275) , however we could switch this around so the settings.yaml is first. I was thinking that the secrets.py should take first preference for backwards compatibility, and then environment should always override everything else (e.g., a config file might already exist but the user can easily set the envar to override it) and then the defaults in settings.py are last. What do you think?

Reply:
Yes, sorry, I read the code wrong. What you have here in `settings.md` matches the behavior in `settings.py`.

It might be better if the env overrides the `settings.yaml` file, but I'm not going to lose any sleep over what you have here.

Reply:
I do agree environment should not be over-ridden - the challenge here is that we don't have a good way to determine if a variable was set in the environment (or is still a default). We can't just check if a DEFAULT[key] is not None, because this won't work for booleans, or generally any variable with a default that isn't None. It's a bit of a catch 22 because in order to set anything from settings.yaml, we need to have all the variables in one lookup, `DEFAULTS`, but we can't set that one variable until we've parsed the different types from the environment! So I think (for now at least) this approach is a reasonable start. If you have a good idea for an implementation though I'd definitely be down to try it!